### PR TITLE
 WFLY-10236 Upgrade Infinispan to 9.2.2.Final

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.jboss.org/browse/WFLY-10236
2nd level cache integration now leverages the infinispan-hibernate-cache-spi module rather using a custom RegionFactory.

### CODE REVIEW

File: jpa/hibernate5-legacy/pom.xml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+  ~  JBoss, Home of Professional Open Source
+  ~  Copyright 2013, Red Hat Middleware LLC, and individual contributors
```
Comment (scottmarlow) [inline jpa/hibernate5-legacy/pom.xml]: should be Copyright 2018.

File: jpa/hibernate5-legacy/src/main/java/org/jboss/as/jpa/hibernate5/infinispan/InfinispanRegionFactory.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2013, Red Hat Middleware LLC, and individual contributors
```
Comment (scottmarlow) [inline jpa/hibernate5-legacy/src/main/java/org/jboss/as/jpa/hibernate5/infinispan/InfinispanRegionFactory.java]: should be Copyright 2018.

Comment (pferraro) [reply jpa/hibernate5-legacy/src/main/java/org/jboss/as/jpa/hibernate5/infinispan/InfinispanRegionFactory.java]: This was an existing file that was moved to a new module - so the original year is correct.

File: clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/PartitionHandlingBuilder.java [hunk 3]
```diff
@@ -49,6 +50,7 @@
     public PartitionHandlingConfiguration getValue() {
         return new ConfigurationBuilder().clustering().partitionHandling()
                 .whenSplit(this.enabled ? PartitionHandling.DENY_READ_WRITES : PartitionHandling.ALLOW_READ_WRITES)
+                .mergePolicy(MergePolicy.NONE)
```
Comment (rvansa) [inline clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/PartitionHandlingBuilder.java]: This is the default.

File: jpa/hibernate5-legacy/src/main/java/org/jboss/as/jpa/hibernate5/infinispan/InfinispanRegionFactory.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ * @author Scott Marlow
+ * @deprecated Use {@link org.infinispan.hibernate.cache.v51.InfinispanRegionFactory} instead in conjunction with "hibernate.cache.infinispan.shared" set to false.
+ */
+@Deprecated
+public class InfinispanRegionFactory extends org.infinispan.hibernate.cache.v51.InfinispanRegionFactory {
+    private static final long serialVersionUID = 6526170943015350422L;
+
+    @Override
+    public void start(SessionFactoryOptions settings, Properties properties) throws CacheException {
+        properties.setProperty("hibernate.cache.infinispan.shared", "false");
```
Comment (rvansa) [inline jpa/hibernate5-legacy/src/main/java/org/jboss/as/jpa/hibernate5/infinispan/InfinispanRegionFactory.java]: use `ManagedEmbeddedCacheManagerProvider.SHARED`?

**Comment by mvinkler** [comment]
QE is OK with the PR.

File: jpa/hibernate5-legacy/src/main/java/org/jboss/as/jpa/hibernate5/infinispan/InfinispanRegionFactory.java (hunk 4 above)
Comment (pferraro) [reply jpa/hibernate5-legacy/src/main/java/org/jboss/as/jpa/hibernate5/infinispan/InfinispanRegionFactory.java]: The module containing these classes intentionally does not reference the module containing ManagedEmbeddedCacheManagerProvider, which requires the jipijapa api, etc.  That's why I'm not using any constants here.
I could externalize this property to a separate file, in a separate module (similar to what you rightly did with InfinispanProperties), but that seemed overkill just for use by a compatibility module that will eventually be dropped.

File: clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/PartitionHandlingBuilder.java (hunk 3 above)
Comment (pferraro) [reply clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/PartitionHandlingBuilder.java]: Yes - this is intentional.  The logic here should match:
https://github.com/infinispan/infinispan/blob/9.2.x/core/src/main/java/org/infinispan/configuration/cache/PartitionHandlingConfigurationBuilder.java#L33

**Pull request merged**